src/dmclock_server.h

```diff
@@ -821,14 +831,23 @@ namespace crimson {
 	RequestTag tag(0, 0, 0, time);
 
 	if (!client.has_request()) {
-	  tag = RequestTag(client.get_req_tag(), client.info,
-			   req_params, time, cost);
+#ifndef DO_DYNAMIC_CLI_INFO_F
```

Comment:
Looking at how to test this thoroughly along with the existing DO_NOT_DELAY_TAG_CALC, since they interact, you'd have to build the code in four different ways with each combination of the settings. For that reason I'm wondering whether if DO_DYNAMIC_CLI_INFO_F would be better as a boolean template parameter to the PriorityQueue classes. Would you please explore that? You don't have to do anything with DO_NOT_DELAY_TAG_CALC, as if you can successfully change DO_DYNAMIC_CLI_INFO_F to a template parameter, I might then make an effort to do the same with DO_NOT_DELAY_TAG_CALC.